api/src/main/java/org/apache/iceberg/io/CloseableIterable.java

```diff
@@ -117,6 +118,16 @@ public O next() {
     };
   }
 
+  static <E> CloseableIterable<E> combine(Iterable<CloseableIterable<E>> iterable,
+                                          ExecutorService workerPool,
+                                          int workerPoolSize) {
+    if (workerPool == null) {
+      return concat(iterable);
+    }
+    Preconditions.checkArgument(workerPoolSize > 0, "Invalid workerPoolSize (not positive): " + workerPoolSize);
```

Comment:
Style: in this project, we use empty lines to separate control flow blocks from the following statements.

I'd also add a new line after the precondition.